src/commonMain/kotlin/org/kosat/CDCL.kt

```diff
@@ -48,96 +92,101 @@ class CDCL(val clauses: MutableList<MutableList<Int>>, initNumber: Int = 0) {
         }
     }
 
-    data class VarState(
-        var status: VarStatus,
-        var clause: Int,
-        var level: Int,
-    )
+    private fun variable(lit: Int): Int = abs(lit)
```

Comment:
Consider wrapping vars/lits in inline classes (in new Kotlin version they are now called `value class`):
```kotlin
// Note: last time I worked with the concept of
// value classes, they were called "inline class".
// So it is better to consult the Kotlin docs for 
// the exact new syntax of value classes.
inline class Lit(private val lit: Int) {
    fun var() -> Var {
        return Var(this.lit << 1)
        // or any other conversion of lit to var,
        // according to your conventions
    }
}